cmake/modules/hunter_install_libsodium_config.cmake

```diff
@@ -0,0 +1,20 @@
+include(CMakePackageConfigHelpers)
+include(hunter_test_string_not_empty)
+
+function(hunter_install_libsodium_config)
+
+    hunter_test_string_not_empty("${HUNTER_libsodium_VERSION}")
+    hunter_test_string_not_empty("${HUNTER_PACKAGE_INSTALL_PREFIX}")
+    hunter_test_string_not_empty("${HUNTER_SELF}")
+
+    set(templates "${HUNTER_SELF}/cmake/templates")
+    set(configs "${HUNTER_PACKAGE_INSTALL_PREFIX}/lib/cmake/libsodium")
+
+    write_basic_package_version_file("${configs}/libsodiumConfig.cmake" VERSION "${HUNTER_libsodium_VERSION}" COMPATIBILITY ExactVersion)
+
+    configure_file(
+            "${templates}/libsodiumConfig.cmake.in"
+            "${configs}/libsodiumConfig.cmake"
```

Comment:
> Is it possible in general? You mean like fat library with multiple architectures?

It is not, there was one of the developers from SDL who worked on a FATELF system to do this with executables in linux, perhaps something similar could be done with libraries but currently it cant

> CMAKE_BUILD_TYPE is not a list variable. For single-configuration generators you have to configure project twice, one for Release and one for Debug. Then you can install them to one directory and they can coexists. Also see: http://stackoverflow.com/a/24470998/2288008

Ohh I see, ok I should be set, I'll figure something out.